jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/Agent.java

```diff
@@ -0,0 +1,126 @@
+    private static final String API_KEY_HEADER = "api-key";
+    private static final String SERVICE_NAME_HEADER = "service.name";
+    private static final String DEFAULT_URL = "http://localhost:4317";
+    private static final String DEFAULT_SERVICE_NAME = "jfr-otlp-bridge";
+    private static final long EXPORT_MILLIS = 2000;
+
+    private static MeterProvider meterProvider;
+
+    public static void agentmain(String agentArgs, Instrumentation inst) {
+      premain(agentArgs, inst);
+    }
+
+    public static void premain(String agentArgs, Instrumentation inst) {
+      meterProvider = GlobalMeterProvider.get();
+      if (meterProvider == null) {
+        configureOpenTelemetry();
+      }
+      startJfrService(agentArgs, inst);
+    }
+
+    public static void startJfrService(String agentArgs, Instrumentation inst) {
+        var jfrMonitorService = Executors.newSingleThreadExecutor();
+        var toMetricRegistry = HandlerRegistry.createDefault(meterProvider);
+
+        jfrMonitorService.submit(() -> {
+            try (var recordingStream = new RecordingStream()) {
+                var enableMappedEvent = eventEnablerFor(recordingStream);
+                toMetricRegistry.all().forEach(enableMappedEvent);
+                recordingStream.setReuse(false);
+                logger.log(Level.FINE, "Starting recording stream...");
+                recordingStream.start(); //run forever
+            }
+        });
+
+    }
+
+    private static Consumer<RecordedEventHandler> eventEnablerFor(RecordingStream recordingStream) {
+        return handler -> {
+            EventSettings eventSettings = recordingStream.enable(handler.getEventName());
+            handler.getPollingDuration().ifPresent(eventSettings::withPeriod);
+            handler.getThreshold().ifPresent(eventSettings::withThreshold);
+            recordingStream.onEvent(handler.getEventName(), handler);
+        };
+    }
+
+    static void configureOpenTelemetry() {
+        // Configure the meter provider
+        var serviceName = System.getenv(SERVICE_NAME);
+        if (serviceName == null) {
+            serviceName = DEFAULT_SERVICE_NAME;
+        }
+        var resource = Resource.getDefault().merge(
+                Resource.create(Attributes.builder()
+                        .put(SERVICE_NAME_HEADER, serviceName)
+                        .build()));
+        var apiKey = System.getenv(API_KEY);
+        var otlpUrl = System.getenv(OTLP_URL);
+        if (otlpUrl == null) {
+            otlpUrl = DEFAULT_URL;
+        }
+
+        var exporterBuilder = OtlpGrpcMetricExporter.builder()
+                .setEndpoint(otlpUrl);
+
+        if (apiKey != null) {
+            exporterBuilder.addHeader(API_KEY_HEADER, apiKey);
+        }
+        var meterProviderBuilder = SdkMeterProvider.builder().setResource(resource);
+        // Set delta aggregation
+        setAggregatorFactory(meterProviderBuilder, COUNTER, Aggregation.sum(DELTA));
```

Comment:
My understanding is generally users should be setting aggregations, not instrumentation. Do we need to set these?

/cc @jsuereth